app/src/main/java/org/p2p/wallet/striga/offramp/ui/StrigaOffRampPresenter.kt

```diff
@@ -156,7 +166,29 @@ class StrigaOffRampPresenter(
     }
 
     override fun onSubmit() {
-        // todo: https://p2pvalidator.atlassian.net/browse/PWN-9267
+        setButtonState(StrigaOffRampButtonState.NextProgress)
+
+        if (strigaUserInteractor.isKycApproved) {
+            launch {
+                try {
+                    // load all necessary data again if it was not loaded before
+                    // enrich crypto + enrich EUR + load statement to extract iban & bic
+                    strigaWalletInteractor.loadDetailsForStrigaAccounts().getOrThrow()
+
+                    // go to withdraw screen
+                    view?.navigateToWithdraw(inputAmountA)
+                } catch (e: Throwable) {
+                    Timber.e(e, "Unable to start Striga withdrawal process")
+                    view?.showUiKitSnackBar(messageResId = R.string.error_general_message)
+                } finally {
+                    setButtonState(StrigaOffRampButtonState.Enabled)
+                }
+            }
+        } else {
+            // go to standard Striga signup flow
+            val destination = strigaUserInteractor.getUserDestination()
+            view?.navigateToSignup(destination)
+        }
```

Comment:
```suggestion
        if (strigaUserInteractor.isKycApproved) {
            navigateToWithdrawFlow()
        } else {
            navigateToSignupFlow()
        }
```

looks better and can remove indentation